plugins/queue/scheduler.js

```diff
@@ -20,6 +20,16 @@ const TEMPORAL_TOKEN_TIMEOUT = 12 * 60; // 12 hours in minutes
 const TEMPORAL_UNZIP_TOKEN_TIMEOUT = 2 * 60; // 2 hours in minutes
 const BLOCKED_BY_SAME_JOB_WAIT_TIME = 5;
 
+/**
+ * Checks whether the job associated with the build is virtual or not
+ * @method isVirtualJob
+ * @param {Object} annotations           Job Annotations
+ * @return {Boolean}
+ */
+function isVirtualJob(annotations) {
+    return annotations && annotations['screwdriver.cd/virtualJob'];
```

Comment:
is this annotation treated as a boolean or string ? If string you might have to check explicitly the expected value

Reply:
Boolean